astroquery/heasarc/core.py

```diff
@@ -135,290 +455,293 @@ def query_object_async(self, object_name, mission, *,
             See :ref:`caching documentation <astroquery_cache>`.
         **kwargs :
             see `~astroquery.heasarc.HeasarcClass._args_to_payload` for list
-            of additional parameters that can be used to refine search query
-        """
-        request_payload = self._args_to_payload(
-            mission=mission,
-            entry=object_name,
-            **kwargs
-        )
+            of additional parameters that can be used to refine search query.
 
-        # Return payload if requested
-        if get_query_payload:
-            return request_payload
+        """
+        pos = coordinates.SkyCoord.from_name(object_name)
+        return self.query_region(pos, table=mission, spatial='cone',
+                                 get_query_payload=get_query_payload)
 
-        return self.query_async(request_payload, cache=cache)
+    def get_links(self, query_result=None, tablename=None):
```

Comment:
are these datalinks? Then call it `get_datalinks` instead (like in gaia), though here were are inconsistent yet again as alma has `datalink` instead. But not many modules have datalinks yet, so we can sort out the API now and use that for the future (and then clean up the inconsistent, but existing modules)

Reply:
We were debating this. If `get_datalinks` is more consistent with other modules, we can switch.